### PR TITLE
[Feature] RC-87 add a query row count button in lineage view

### PR DESCRIPTION
**Pull request by kentwelcome**
**PR checklist**

- [x] Ensure you have added or ran the appropriate tests for your PR.
- [x] DCO signed

**What type of PR is this?**
Feature

**What this PR does / why we need it**:
- Add `Query Row Count` button on the Lineage View page
  CleanShot 2023-12-26 at 15 02 26

- When users click that button, we will query all the row counts of showing models individually.
  CleanShot 2023-12-26 at 15 03 07

- Refactor row count tag component

**Which issue(s) this PR fixes**:
RC-87

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

### CODE REVIEW

File: js/src/components/lineage/GraphNode.tsx [hunk 1]
```diff
@@ -102,6 +103,14 @@ export function GraphNode({ data }: GraphNodeProps) {
               </Flex>
             )}
           </Flex>
+          {data.resourceType === "model" && (
+          <Flex flex='1 0 auto' mx="1" direction="column" paddingBottom="1">
+            <HStack spacing={"8px"}>
+              <Spacer />
+              <RowCountTag node={data} isInteractive={false}/>
```
Comment (popcornylu) [inline js/src/components/lineage/GraphNode.tsx:115]: Don't show the row count tag if it is zoom out

image

File: js/src/components/lineage/NodeTag.tsx [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+  }
+
+  if (isInteractive === false && isFetched === false && isFetching === false) {
+    // Don't show anything if the row count is not fetched and is not interactive.
+    return null;
+  }
+
+
+  return (
+    <Tooltip hasArrow label={isFetched || isFetching || !isInteractive ?"Number of row":"Query the number of row"}>
```
Comment (popcornylu) [inline js/src/components/lineage/NodeTag.tsx]: It seems very easy to open the tooltip. Could we add a `openDelay`?
https://chakra-ui.com/docs/components/tooltip/props

image

File: js/src/components/lineage/NodeTag.tsx [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+  useEffect(() => {
+    if (isFetched) {
+      if (index + 1 < nodes.length) {
+        // TODO: Use BFS to walk though all the changed nodes first.
+        setIndex(index + 1);
+      } else{
+        setEnabled(false);
+      }
+    }
+  }, [isFetched, index, nodes]);
```
Comment (popcornylu) [inline js/src/components/lineage/NodeTag.tsx:172]: [Recommendataion]

Here is a bit tricky. In `useQuery`, you have more options to do this

1. useQueries
2. useQueryClient + [setQueryData](https://tanstack.com/query/v4/docs/react/reference/QueryClient#queryclientsetquerydata)

File: js/src/components/lineage/GraphNode.tsx (hunk 1 above)
Comment (kentwelcome) [reply js/src/components/lineage/GraphNode.tsx:115]: fix

File: js/src/components/lineage/NodeTag.tsx (hunk 2 above)
Comment (kentwelcome) [reply js/src/components/lineage/NodeTag.tsx]: fix

File: js/src/components/lineage/NodeTag.tsx (hunk 3 above)
Comment (kentwelcome) [reply js/src/components/lineage/NodeTag.tsx:172]: I had considered using `useQueries` before. However, the `useQueries` in TanStack Query doesn't provide a way to limit the query tasks. When users try to query lots of models and the warehouse response time is slow, it's very easy to hit query timeout.
I have created a story (RC-95) to enhance the query algorithm. We can enhance the query code in that story.

Comment (popcornylu) [reply js/src/components/lineage/NodeTag.tsx:172]: So i prefer to use option 2. Yes. please use another story (RC-95) to enhance it

**Review by popcornylu — APPROVED** [review]
LGTM

**Pull request merged**